In [1]:
# Importing libraries and reading dataframe
import pandas as pd
import numpy as np

df = pd.read_csv("../raw-data/data_rearrangement.csv").dropna(how='all', axis='columns')
df.head(9)

Unnamed: 0  Shaking  Flipping
0     walking_hand       25         0
1      walking_bag       22         5
2           pocket       15         5
3              bag       12         0
4    passing_phone        6         4
5  hold_phone_flat        3        12
6           selfie        9         7
7   dropping_phone       11         3

In [2]:
# Creating a third column to compute the difference between the num of shaking detected and flipping
df["Difference"] = df['Shaking'] - df['Flipping']
df.head(9)

Unnamed: 0  Shaking  Flipping  Difference
0     walking_hand       25         0          25
1      walking_bag       22         5          17
2           pocket       15         5          10
3              bag       12         0          12
4    passing_phone        6         4           2
5  hold_phone_flat        3        12          -9
6           selfie        9         7           2
7   dropping_phone       11         3           8

In [3]:
# We'll essentially work with the difference column to do our t-test

mean_difference = df['Difference'].mean()
sample_std_difference = df['Difference'].std()

print(f"The mean of differences is {mean_difference} and its standard deviation is {sample_std_difference:.4f}")

The mean of differences is 8.375 and its standard deviation is 10.3501


In [4]:
t_value = (mean_difference-0)/(sample_std_difference/np.sqrt(8))
critical_t = 1.895
print(f"t_value = {t_value:.4f}")
print(f"critical_t_value is {critical_t:4f}")

t_value = 2.2887
critical_t_value is 1.895000


* Our H0 here would be that the difference of activation times of the 2 gestures is 0; that is, the number of activation is equal for both gestures and that there's not robustness difference between them.

* Our alternative hyphothesis HA would be that there is actual difference between the number of activations.

We are doing a one sided right tail test, which indicates that the (n shaking gesture) - (n facedown gesture) > 0. Which is the same as saying that the shaking gesture will be activated more times that the face down gesture and hence the face down gesture is more robust to accidental activation. 

The t-value is computed as: mean/(std/√n), where n is in our case 8. t_value = 2.2887.

The critical t value is found by looking at the t-value table. With confidence interval 5 and degrees of freedom (8-1=)7 for a one tailed t-test, our critical_t_value is 1.8950.

It can be observed that our t-value is greater than the critical t-value and hence we reject our null hypothesis. Moreover, it can be seen that the p-value (P(T>our t-value)) has a low value (<<0.1) and hence we can be quite sure that the null hypothesis should be rejected. Up to this point we can say the robustness of the 2 gestures is different and we need to determine which is more robust to accidental activation. Since we have procceded to do the right tail test, we **can** conclude that the flip down gesture is **more** robust to actidental activation and with this we finish our proof.